### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'findController')

### DISCUSSION THREAD

**Discussion by rmdorsey** [stephanrauh/ngx-extended-pdf-viewer#2508] [Q&A] [answered]
I'm attempting to implement the find by regex feature as described here: https://pdfviewer.net/extended-pdf-viewer/custom-find

In the showcase code it shows an override method that appears to be attempting to access the finder on the view. 
![image](https://github.com/user-attachments/assets/d86bbedd-c439-4ad2-8de1-57041a17e434)

I'm attempting to access the viewer like this:
![image](https://github.com/user-attachments/assets/60e32746-f068-4ec9-967d-69c23bf2abf9)

In previous versions this worked, but with the newest version it's showing this error in the console:
![image](https://github.com/user-attachments/assets/4e924f80-1f56-4842-b3f7-8f67b08ac88f)

Please advise.  Thanks!

**Comment by stephanrauh** [comment, marked as answer]
Currently I'm implementing a better solution, but in the meantime, let's fix your bug. Luckily, that's easy. With version 21 I've removed `PDFViewerApplication` from the `window` object. Polluting the global namespace was a dirty hack that has severe disadvantages.

Nowadays, you can use the `PDFNoticationService` to get the instance of the `PDFViewerApplication`:

```ts
constructor(notificationService: PDFNotificationService) {
    effect(() => this.PDFViewerApplication = notificationService.onPDFJSInitSignal());
}
```

Update: I've added the missing constructor to the showcase page.

**Reply by rmdorsey** [reply]
Yep!  That works!  Thanks!
